code/datums/mutantraces.dm

```diff
@@ -493,6 +493,124 @@
 
 	say_filter(var/message)
 		return replacetext(message, "s", stutter("ss"))
+		
+/datum/mutantrace/doggo
+	name = "doggo"
+	// icon_state = "doggo"
+	icon_state = "doggo"
+	icon_override_static = 1
+	allow_fat = 1
+	override_attack = 0
+	voice_override = "doggo"
+
+	New(var/mob/living/carbon/human/H)
+		..()
+		if(ishuman(mob))
+			var/datum/appearanceHolder/aH = mob.bioHolder.mobAppearance
+
+			detail_1 = image('icons/effects/genetics.dmi', icon_state="doggo_detail-1", layer = MOB_LIMB_LAYER+0.1)
+			detail_2 = image('icons/effects/genetics.dmi', icon_state="doggo_detail-2", layer = MOB_LIMB_LAYER+0.2)
+			detail_3 = image('icons/effects/genetics.dmi', icon_state="doggo_detail-3", layer = MOB_LIMB_LAYER+0.3)
+			detail_over_suit = image('icons/effects/genetics.dmi', icon_state="doggo_over_suit", layer = MOB_LAYER_BASE+0.3)
+
+			hex_to_rgb_list(aH.customization_first_color)
+
+			detail_1.color = fix_colors(aH.customization_first_color)
+			detail_2.color = fix_colors(aH.customization_second_color)
+			detail_3.color = fix_colors(aH.customization_third_color)
+			detail_over_suit.color = fix_colors(aH.customization_first_color)
+
+			// detail_1.color = aH.customization_first_color
+			// detail_2.color = aH.customization_second_color
+			// detail_3.color = aH.customization_third_color
+
+			mob.update_face()
+			mob.update_body()
+			mob.update_clothing()
+
+	proc/fix_colors(var/hex)
+		var/list/L = hex_to_rgb_list(hex)
+		for (var/i in L)
+			L[i] = min(L[i], 250)
+			L[i] = max(L[i], 5)
+		if (L.len == 3)
+			return rgb(L["r"], L["g"], L["b"])
+		return rgb(22, 210, 22)
+
+	sight_modifier()
+		mob.see_in_dark = SEE_DARK_HUMAN + 1
+		mob.see_invisible = 1
```

Comment:
I think this should be called out in the PR description. Looks like they have better ability to see in the dark than humans, and I'm not entirely sure what `see_invisible` does but I think it might be the same way that cyborgs see people using a cloaking device?

Reply:
It was copied from the lizard mutantrace, and as far as I can tell it doesn't allow them to see cloakers.

...it would explain why lizards are able to see bloodlings, come to think of it.

I'll add that to the PR description.

Reply:
Looking into a bit more now that I'm able to, I don't really know how `see_invisible` is derived. It seems to be some sort of bitflag but also "gradually increasing amounts let you see more invisible things", so yeah, I'd want someone who knows a bit more about this to comment (or you can do the investigation yourself/remove it for the time being).